msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ServiceFabricManagedIdentitySource.java

```diff
@@ -41,12 +39,9 @@ public void createManagedIdentityRequest(String resource) {
         managedIdentityRequest.queryParameters.put("resource", Collections.singletonList(resource));
         managedIdentityRequest.queryParameters.put("api-version", Collections.singletonList(SERVICE_FABRIC_MSI_API_VERSION));
 
-        if (idType == ManagedIdentityIdType.CLIENT_ID) {
-            LOG.info("[Managed Identity] Adding user assigned client id to the request for Service Fabric Managed Identity.");
-            managedIdentityRequest.queryParameters.put(Constants.MANAGED_IDENTITY_CLIENT_ID, Collections.singletonList(userAssignedId));
-        } else if (idType == ManagedIdentityIdType.RESOURCE_ID) {
-            LOG.info("[Managed Identity] Adding user assigned resource id to the request for Service Fabric Managed Identity.");
-            managedIdentityRequest.queryParameters.put(Constants.MANAGED_IDENTITY_RESOURCE_ID, Collections.singletonList(userAssignedId));
+        if (this.idType != null && !StringHelper.isNullOrBlank(this.userAssignedId)) {
+            LOG.info("[Managed Identity] Adding user assigned ID to the request for Service Fabric Managed Identity.");
```

Comment:
This does reduce the visibility in logs as to which user assigned is being used.

Reply:
Have a look at the changes in the `ManagedIdentityRequest` class: the new `addUserAssignedIdToQuery` method called right after this log has the same sort of "Adding [this type of ID] to the request" log message that these classes used to have.

You'll still get the same information about the MI source type and ID type being used, it's just that they will be in two info logs.